source/opt/common_uniform_elim_pass.cpp

```diff
@@ -301,6 +327,10 @@ bool CommonUniformElimPass::UniformAccessChainConvert(ir::Function* func) {
         continue;
       if (HasUnsupportedDecorates(ptrInst->result_id()))
         continue;
+      if (IsVolatileVar(varId))
+        continue;
+      if (IsVolatileLoad(&*ii))
```

Comment:
Since "IsVolatileLoad" is a less expensive check, it might be useful to move it up.  No point walking all of the decorations if this check will fail anyway.